core/src/datayoga_core/blocks/relational/write/block.py

```diff
@@ -140,7 +139,7 @@ def execute_upsert(self, records: List[Dict[str, Any]]):
             # map the record to upsert based on the mapping definitions
             # add nulls for missing mapped fields
             record_to_upsert = {}
-            for item in self.mapping:
+            for item in self.keys + self.mapping:
```

Comment:
you need to handle duplication case here where the column in both in mapping and in keys

Reply:
or decide that in such a case you just take the one from mapping (and just add that to the block's doc)

Reply:
opened #214 for this.